engine/access/rpc/engine.go

```diff
@@ -90,6 +92,7 @@ func New(log zerolog.Logger,
 	rpcMetricsEnabled bool,
 	apiRatelimits map[string]int, // the api rate limit (max calls per second) for each of the Access API e.g. Ping->100, GetTransaction->300
 	apiBurstLimits map[string]int, // the api burst limit (max calls at the same time) for each of the Access API e.g. Ping->50, GetTransaction->10
+	proxy *observer.FlowCachedAccessAPI,
```

Comment:
Can we use [functional options](https://golang.cafe/blog/golang-functional-options-pattern.html#:~:text=The%20Go%20(Golang)%20Functiona%20Options,and%20initialisation%20of%20your%20struct.) pattern here instead?

Reply:
Yes, there are many of these and this is optional. I will probably get to it only tomorrow.

Reply:
Fair enough.
Addressed: https://github.com/GetElastech/flow-go/commit/a24c288af5289a0cf9f40db4d77dbecc95cbcba3
I am working on a more extensive solution, but I did not want to block you with that one.
How about using a builder? It takes up less code, unless you have automation that shuffles the options.

Reply:
Addressed. Comments inside.
Thank you for the review Simon.